### PR TITLE
[Test Kitchen] Cleans up some lingering kitchen errors

### PR DESCRIPTION
**Pull request by gmmeyer**
### What does this PR do?

There were two small issues to fix in the test kitchen:

First, there was a race condition that this should help solve for in installing the agent. Since the chef install does not wait on the agent install being finished, it can lead to a race condition with the tests.

Second, it creates a random password, making it more secure than using the default one. (And it fixes an issue that cropped up with using the default username.)

### Motivation

I wanted the test kitchen to work!

### Additional Notes

I really hope there is no more work with the test kitchen.

### CODE REVIEW

File: test/kitchen/tasks/run-test-kitchen.sh [hunk 1]
```diff
@@ -82,6 +82,11 @@ if [ -z ${AGENT_VERSION+x} ]; then
   popd
 fi
 
+# Generate a password to use for the windows servers
+if [ -z ${SERVER_PASSWORD+x} ]; then
+  export SERVER_PASSWORD=$(date +%s | sha256sum | base64 | head -c 32 ; echo)
```
Comment (olivielpeau) [inline test/kitchen/tasks/run-test-kitchen.sh]: this password is only determined by the unix epoch (in seconds) at which the command was run. Could we use `/dev/urandom` directly instead? Something like `< /dev/urandom tr -dc A-Za-z0-9 | head -c32` should work

File: test/kitchen/test/integration/common/rspec/spec_helper.rb [hunk 2]
```diff
@@ -197,7 +197,15 @@ def is_port_bound(port)
...
-      msi_path = 'C:\\Users\\azure\\AppData\\Local\\Temp\\kitchen\\cache\\ddagent-cli.msi'
+      msi_path_base = 'C:\\Users\\'
+      msi_path_end = '\\AppData\\Local\\Temp\\kitchen\\cache\\ddagent-cli.msi'
+      msi_path_azure = msi_path_base + 'azure' + msi_path_end
+      msi_path_datadog = msi_path_base + 'datadog' + msi_path_end
+      if File.exist?(msi_path_azure)
+        msi_path = msi_path_azure
+      else
+        msi_path = msi_path_datadog
+      end
```
Comment (olivielpeau) [inline test/kitchen/test/integration/common/rspec/spec_helper.rb]: what's this block of changes for? (a comment to explain why we do this would be perfect)

**Review by olivielpeau — COMMENTED** [review]
Thanks for cleaning this up! Added a couple of comments, other than that LGTM!

File: test/kitchen/test/integration/common/rspec/spec_helper.rb (hunk 2 above)
Comment (gmmeyer) [reply test/kitchen/test/integration/common/rspec/spec_helper.rb]: Sure! Basically I changed the username when I made a better password, the one I had before was azure, and I wanna make this more fault tolerant by also allowing the default username to be used without it breaking. The error message is really horrible and hard to parse, so I wanna avoid anyone having to parse it.

File: test/kitchen/tasks/run-test-kitchen.sh (hunk 1 above)
Comment (gmmeyer) [reply test/kitchen/tasks/run-test-kitchen.sh]: Sure, that's better

**Review by derekwbrown — DISMISSED** [review]

**Review by olivielpeau — APPROVED** [review]
:shipit:
